extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelHelper.java

```diff
@@ -65,25 +67,44 @@ public static String resolveChannelId(@Nonnull final String idOrPath)
         // URL, then no information about the channel associated with this URL was found,
         // so the unresolved url will be returned.
         if (!channelId[0].equals("channel")) {
-            final byte[] body = JsonWriter.string(
-                    prepareDesktopJsonBuilder(Localization.DEFAULT, ContentCountry.DEFAULT)
-                            .value("url", "https://www.youtube.com/" + idOrPath)
+            String urlToResolve = "https://www.youtube.com/" + idOrPath;
+
+            JsonObject endpoint = new JsonObject();
+            String webPageType = "";
+            // Try to resolve YT channel redirects
+            // It works like that:
+            // @TheDailyShow
+            // -> resolves to thedailyshow
+            // -> resolves to the id: UCwWhs_6x42TyRM4Wstoq8HA
+            // Please note that this is not always the case, some handles
+            // e.g. @google or @Gronkh directly resolve the id
+            for (int tries = 0;
+                 urlToResolve != null && tries < 3;
+                 tries++) {
+                final byte[] body = JsonWriter.string(
+                        prepareDesktopJsonBuilder(Localization.DEFAULT, ContentCountry.DEFAULT)
+                            .value("url", urlToResolve)
                             .done())
                     .getBytes(StandardCharsets.UTF_8);
 
-            final JsonObject jsonResponse = getJsonPostResponse(
+                final JsonObject jsonResponse = getJsonPostResponse(
                     "navigation/resolve_url", body, Localization.DEFAULT);
 
-            checkIfChannelResponseIsValid(jsonResponse);
+                checkIfChannelResponseIsValid(jsonResponse);
 
-            final JsonObject endpoint = jsonResponse.getObject("endpoint");
+                endpoint = jsonResponse.getObject("endpoint");
 
-            final String webPageType = endpoint.getObject("commandMetadata")
+                webPageType = endpoint.getObject("commandMetadata")
                     .getObject("webCommandMetadata")
                     .getString("webPageType", "");
 
-            final JsonObject browseEndpoint = endpoint.getObject(BROWSE_ENDPOINT);
-            final String browseId = browseEndpoint.getString(BROWSE_ID, "");
+                urlToResolve = "WEB_PAGE_TYPE_UNKNOWN".equals(webPageType)
+                    ? endpoint.getObject("urlEndpoint").getString("url")
+                    : null;
+            }
+
+            final String browseId = endpoint.getObject(BROWSE_ENDPOINT)
+                .getString(BROWSE_ID, "");
 
             if (webPageType.equalsIgnoreCase("WEB_PAGE_TYPE_BROWSE")
                     || webPageType.equalsIgnoreCase("WEB_PAGE_TYPE_CHANNEL")
```

Comment:
Btw this (|| &&) combination might be wrong, since clarifying parenthesis are missing. What is the intended condition here @AudricV?

Reply:
the ``equalsIgnoreCase`` should also likely be inverted for better null safety (we can also then remove the ``""`` initialization above)

Reply:
I wonder if there's any checkstyle rule for this, but I could only find one which does the opposite :sweat_smile: https://checkstyle.org/checks/coding/unnecessaryparentheses.html